### PR TITLE
Make `squared_difference` alias of `squared_error `

### PR DESCRIPTION
**Pull request by hvy**
`squared_difference` seems identical to `squared_error`, so I aliased one of them. Documentation might need to be fixed as well 🤔

### CODE REVIEW

File: chainer/functions/loss/squared_error.py [hunk 1]
```diff
@@ -49,5 +49,19 @@ def squared_error(x0, x1):
...
+        array([0., 1., 2., 3., 4., 5.], dtype=float32)
+        >>> x2 = np.array([5, 4, 3, 2, 1, 0]).astype(np.float32)
+        >>> x2
+        array([5., 4., 3., 2., 1., 0.], dtype=float32)
+        >>> y = F.squared_error(x1, x2)
+        >>> y.shape
+        (6,)
+        >>> y.array
+        array([25.,  9.,  1.,  1.,  9., 25.], dtype=float32)
+
```
Comment (hvy) [inline chainer/functions/loss/squared_error.py]: Carried over example code from `squared_difference`. Should make users more happy.

**Comment by Crissman** [comment]
Docs look fine.

Screen Shot 2019-06-19 at 4 10 00 PM

**Comment by Crissman** [comment]
Don't repeat your tests. I think the test for `squared_difference` should be removed as well.

`tests/chainer_tests/functions_tests/math_tests/test_squared_difference.py`

**Comment by niboshi** [comment]
How about leaving `def squared_difference()` and its docstring?

**Comment by Crissman** [comment]
ChainerX has squared_difference defined in `chainerx_cc/chainerx/python/routines.cc`, but squared_error does not appear to be defined. I think Chainer and ChainerX should have the same base. Should this be added?

**Comment by Crissman** [comment]
> How about leaving `def squared_difference()` and its docstring?

Not sure it's worth leaving the file and the definition for that minor issue. I think it's ok to delete.

**Comment by hvy** [comment]
I think it's fine to delete it too. As for the tests, they should probably not be removed as the public interface is not reconsidered by any means. For the ChainerX routine, I'm not sure if both of these names were considered at the time it was implemented but it's okay to provide either of them (as it currently does)?

**Comment by niboshi** [comment]
Reader will be confused if `F.squared_error` appears in the example code of `F.squared_difference`.

**Comment by hvy** [comment]
What do you think of adding a note, maybe before the example, that they're identical?

**Comment by niboshi** [comment]
That might be ok, but I think it's much better to leave `F.squared_difference`'s own docstring.

**Comment by hvy** [comment]
Added a note and a comment to the example to reduce confusion.

**Comment by niboshi** [comment]
What if we want to deprecate `F.squared_difference`?
I still think it will make things simpler to just leave the docstring of `F.squared_difference` and I don't see any strong motivation to delete it.

**Comment by hvy** [comment]
Added the definition of `squared_difference` back for reasons mentioned above. Simplified tests and docs for easier maintenance.

File: chainer/functions/loss/squared_error.py [hunk 2]
```diff
@@ -49,5 +49,42 @@ def squared_error(x0, x1):
             A variable holding an array representing the squared error of
             two inputs.
 
+    .. note::
+
+        :func:`~chainer.functions.squared_error` and
+        :func:`~chainer.functions.squared_difference` are identical functions.
```
Comment (Crissman) [inline chainer/functions/loss/squared_error.py]:
```suggestion
        :func:`~chainer.functions.squared_difference` are identical functions,
```

File: chainer/functions/loss/squared_error.py [hunk 3]
```diff
@@ -49,5 +49,42 @@ def squared_error(x0, x1):
...
+        Only with different argument names.
```
Comment (Crissman) [inline chainer/functions/loss/squared_error.py]:
```suggestion
        aside from the different argument names.
```

**Comment by hvy** [comment]
Thank you for the suggestions, just applied them.

**Comment by Crissman** [comment]
Jenkins, test this please.

**Comment by Crissman** [comment]
Jenkins failure looks unrelated.

**Pull request merged**